kie-server-parent/kie-server-services/kie-server-services-task-assigning/kie-server-services-task-assigning-planning/src/main/java/org/kie/server/services/taskassigning/planning/SolutionChangesBuilder.java

```diff
@@ -267,6 +298,54 @@ private void addTaskChanges(final Task task,
         }
     }
 
+    private void addUserChanges(final Map<String, User> usersById,
+                                final List<AddUserProblemFactChange> newUserChanges,
+                                final List<ProblemFactChange<TaskAssigningSolution>> updateUserChanges) {
+
+        final List<User> updatedUserList = filterDuplicates(externalUserList)
+                .filter(externalUser -> !IS_PLANNING_USER.test(externalUser.getId()))
+                .map(UserUtil::fromExternalUser)
+                .collect(Collectors.toList());
```

Comment:
You can avoid creating the `updatedUserList` if you move the `for` cycle below into a `forEach()` here.